### PR TITLE
add config option

### PR DESCRIPTION
**Pull request by AaronNBrock**
This PR allows you to specify config location either by the `--config` cli option or the `GATE_CONFIG` environment variable.

As requested by #24

### CODE REVIEW

File: cmd/gate/root.go [hunk 1]
```diff
@@ -47,22 +48,35 @@ func Execute() {
...
+	} else {
+		v.SetConfigName("config")
+		v.AddConfigPath(".")
+	}
+
 	// If a config file is found, read it in.
-	if err := v.ReadInConfig(); err == nil {
+	err := v.ReadInConfig()
+	if err != nil {
+		fmt.Println(fmt.Errorf("Error reading config file: %s \n", err))
```
Comment (robinbraemer) [inline cmd/gate/root.go:78]: Gate supports configuration by env variables and does not require a config file.
Could you please ignore the error and maybe add a comment that a config file is not required?

**Review by robinbraemer — CHANGES_REQUESTED** [review]
Thank you for your PR! Please see my review comment.

File: cmd/gate/root.go (hunk 1 above)
Comment (AaronNBrock) [reply cmd/gate/root.go:78]: I realized that at one point... but then I forgot _doh!_

Comment (AaronNBrock) [reply cmd/gate/root.go:78]: Counterpoint:  What about situations where the error is due to malformed configuration files?  That seems useful, I think only the "not found" should be ignored.

File: cmd/gate/root.go [hunk 2]
```diff
@@ -65,18 +66,20 @@ func initConfig() {
...
-	err := v.ReadInConfig()
-	if err != nil {
-		fmt.Println(fmt.Errorf("Error reading config file: %s \n", err))
-	} else {
+	switch err := v.ReadInConfig(); err.(type) {
+	case nil:
 		fmt.Println("Using config file:", v.ConfigFileUsed())
+	case viper.ConfigFileNotFoundError: // Do Nothing (config is optional)
+	default:
+		fmt.Println(fmt.Errorf("Error reading config file: %s \n", err))
```
Comment (robinbraemer) [inline cmd/gate/root.go:82]: The problem would be that Gate would start nevertheless, ignoring this error, which is not good if one has explicitly set the flag.

**Comment by robinbraemer** [comment]
I've reworked all this in https://github.com/minekube/gate/commit/a93279cee5793dcef5cf31d0cf3395d5f7b4769a
and thank you very much for pointing out this critical issue!

**Pull request closed without merging**